tests/views/helpers.py

```diff
@@ -0,0 +1,79 @@
+import dash
+from selenium.common.exceptions import TimeoutException
+
+from tests.conftest import select_first
+
+
+def setup_plugin(dash_duo, name, pluginClass, window_size=(1024, 3600)):
+    app = dash.Dash(name)
+    plugin = pluginClass(app, project_identifier=None)
+    layout = plugin.layout
+    app.layout = layout
+    dash_duo.start_server(app)
+    windowsize = window_size
+    dash_duo.driver.set_window_size(*windowsize)
+    return plugin
+
+
+def select_ensemble(dash_duo, plugin, wanted_ensemble_name=None):
+    """tries to select, i.e. click, the ensemble given by ensemble_name, and
+    clicks the first one if no name is given. It returns the name of the
+    selected ensemble."""
+    ensemble_selector_id = plugin.uuid("ensemble-multi-selector")
+    if not wanted_ensemble_name:
+        first_ensemble_name = select_first(dash_duo, "#" + ensemble_selector_id)
+        dash_duo.wait_for_contains_text(
+            "#" + plugin.uuid("selected-ensemble-dropdown"),
+            first_ensemble_name,
+            timeout=4,
+        )
+        return first_ensemble_name
+
+    ensemble_selectors = dash_duo.find_elements(f"#{ensemble_selector_id} option")
+    ensemble_selector = list(
+        filter(
+            lambda ensemble_element: ensemble_element.text == wanted_ensemble_name,
+            ensemble_selectors,
+        )
+    )[0]
+
+    ensemble_selector.click()
+    dash_duo.wait_for_contains_text(
+        "#" + plugin.uuid("selected-ensemble-dropdown"),
+        wanted_ensemble_name,
+        timeout=4,
+    )
+
+
+def select_response(dash_duo, plugin, response_name):
+    _select_param_or_response(
+        dash_duo,
+        plugin,
+        f'#{plugin.uuid("parameter-selector-multi-resp")}',
+        response_name,
+    )
+
+
+def _select_param_or_response(dash_duo, plugin, containerSelector, option_name):
+    selectors = dash_duo.find_elements(f"{containerSelector} option")
+    wanted_option_selector = list(
+        filter(lambda rs_sel: rs_sel.text == option_name, selectors)
+    )[0]
+    wanted_option_selector.click()
+    dash_duo.wait_for_element(f"#{plugin.uuid(option_name)}")
+
+
+def select_parameter(dash_duo, plugin, parameter_name):
+    _select_param_or_response(
+        dash_duo,
+        plugin,
+        f'#{plugin.uuid("parameter-selector-multi-param")}',
+        parameter_name,
+    )
+
+
+def wait_a_bit(dash_duo, time_seconds=0.1):
+    try:
+        dash_duo.wait_for_element(".foo-baarrrrr", timeout=time_seconds)
+    except TimeoutException:
+        pass
```

Comment:
As an alternative to a separate PR it would also be ok to me at least having separate commits in the same PR. One solving the actual issue and then the refactoring afterwards. And yes - i fully agree to your comment keeping the changes needed to solve the PR to a minimum :+1: :+1:

Reply:
@DanSava - thanks for the pointer with the `helper` file naming, I'm very much not versed in the coding style and best practice when it comes to python testing. fixtures, I'll have to read a bit more about, but i have a feeling it's not super relevant in this case?
of course I can move these functions to the conftest file. can a conftest file reside at different scopes (in different directories), or can there only be one?

also - if you have any links to best practice for testing with fixtures and conftest files and whatnot flying around / in your history, would you mind sharing some?

You're right of course, I see now that I went a bit overboard with applying these helper functions outside of the code I added - mea culpa. I'm gonna do these changes as a clear-cut separate commit, as @oysteoh suggests (thanks for that!)

Reply:
after reading a bit more of the code in `tests/conftest.py`, I see that the helper functions i wrote are somewhat similar to `select_first` and `select_by_name` - should I try to refactor my helpers so they use those? I guess that would reduce the complexity and make the code overall DRYer

Reply:
ah `select_by_name` is a new function : )

Reply:
> of course I can move these functions to the conftest file. can a conftest file reside at different scopes (in different directories), or can there only be one?


Yes a conftest file can reside at different scopes, also there can be multiple of these files more info here : https://docs.pytest.org/en/6.2.x/fixture.html#conftest-py-sharing-fixtures-across-multiple-files
But for a project as complex as webviz-ert one conftest file is more than enough. 


> after reading a bit more of the code in tests/conftest.py, I see that the helper functions i wrote are somewhat similar to select_first and select_by_name - should I try to refactor my helpers so they use those? I guess that would reduce the complexity and make the code overall DRYer


Yes, I think it is a good idea to try and use as much as possible the already implemented used and tested logic that is present before jumping and adding a bunch of new logic that might just be similar.

Reply:
hehe, I realized I was already using `find_first`, and we basically did the exact same thing in the other helper function ^^ that was an easy refactor!